lib/package_utils.rb

```diff
@@ -2,12 +2,35 @@
 # Utility functions that take either a package object or a component of a package object as primary input.
 require 'json'
 require_relative 'const'
+require_relative 'convenience_functions'
 require_relative 'downloader'
 
 class PackageUtils
+  def self.reverse_dependency_lookup(dep_name, installed: false)
+    device_json       = ConvenienceFunctions.load_symbolized_json
+    pkgs_that_need_it = []
+
+    if installed
+      device_json[:installed_packages].each do |installed_pkg_info|
```

Comment:
```suggestion
    pkgs_that_need_it = []

    if installed
      ConvenienceFunctions.load_symbolized_json[:installed_packages].each do |installed_pkg_info|
```

or something to that effect? 

at the very least, since we only use `device_json` conditionally, lets only load it conditionally.

not using an intermediate variable at all is more of a style preference, and i don't really mind either way